rnd/autogpt_builder/src/hooks/useAgentGraph.ts

```diff
@@ -637,11 +643,8 @@ export default function useAgentGraph(
         nodes: formattedNodes,
         links: links,
       };
-
-      if (savedAgent && deepEquals(payload, savedAgent)) {
-        console.debug(
-          "No need to save: Graph is the same as version on server",
-        );
+      if (savedAgent && deepEquals(payload, savedAgent, true)) {
```

Comment:
What keys are being added on save that we need the `allowMissingKeys`?

Reply:
Any random field being added from the server can break this.
Currently:
id, subgraph, version,

Reply:
Also additional fields on nodes too I don't really remember what those are

Reply:
@Pwuts I think it's safer to just explicitly define what needs to be compared, I have reverted this change here:
https://github.com/Significant-Gravitas/AutoGPT/pull/8054/files#diff-7e0162cb7a3b78d205d099d20b98cdf25239e6faa2113d6843520a8e21797296R33-R34